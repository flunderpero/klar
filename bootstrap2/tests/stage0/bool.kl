fn main():
    let a = false
    let b = true
    print("`bool` type:")
    print(f"true: {a}")
    print(f"false: {b}")
    print(f"a == b: {a == b}")
    print(f"a == a: {a == a}")
    print(f"b == a or b == b: {b == a or b == b}")
    print(f"b == a or a == b: {b == a or a == b}")
    print(f"b == a and a == b: {b == a and a == b}")
    print(f"a == a and b == b: {a == a and b == b}")
    print("`str` type:")
    let sa = "a"
    let sb = "b"
    print(f"sa == sb or sa == sa: {sa == sb or sa == sa}")
    print("precedence:")
    print(f"true or false and false: {true or false and false}")
end
