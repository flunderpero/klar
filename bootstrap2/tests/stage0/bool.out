`bool` type:
true: false
false: true
a == b: false
a == a: true
b == a or b == b: true
b == a or a == b: false
b == a and a == b: false
a == a and b == b: true
`str` type:
sa == sb or sa == sa: true
precedence:
true or false and false: true
